### DISCUSSION TITLE
Lost all of my data during a BTCPay Server version downgrade (hosted on myNode)

### DISCUSSION THREAD

**Discussion by brettjdicioccio** [btcpayserver/btcpayserver#3556] [Technical Support]
I was hosting a BTCPay Server on myNode (a Bitcoin node platform running on a RaspPi 4 8GB). It contained many invoices and a hot wallet. myNode released a new update that upgraded BTCPay Server to a newer version. This caused a 502 Bad Gateway error when attempting to access the BTCPay Server website. I am assuming the cause of this is an update to something in the BTCPay Server configuration that myNode isn’t accounting for. As part of the troubleshooting process I downgraded my BTCPay Server version and was able to access the website again.

However, there is one big problem. It is starting me from scratch, asking me to create an account for the first time as if it is a completely new BTCPay Server (which it is). I suppose I lost all of my invoices and for the hot wallet funds I will have to restore the recovery phrase somewhere else to retrieve them. I have absolutely no clue what myNode did during the downgrade process. I’m assuming it destroyed the docker container and reset the Postgres database without warning which is terrible.

**The main reason for this post:**
1. Is the data from my previous BTCPay Server instance most likely gone gone? Should I even bother looking to recover it?
2. Is there anything I can do in the Linux terminal to check Postgres or look for a backup that might have been taken somewhere during the downgrade.

This really sucks because when you think “downgrade” you think you are just changing versions and not going to lose all of your data. But with the BTCPay Server architecture using docker for both Postgres and the Website, maybe this isn’t the case.

**Comment by thecoldcarrot** [comment]
I am having a similar issue. What version did you downgrade to in order to not receive the 502 error?

**Comment by brettjdicioccio** [comment]
BTCPAYSERVER_VERSION=“1.3.7”

NOTE: I lost all of my data when doing this. Please make sure you take a
backup first.

**Reply by dennisreimann** [reply]
First of all: Sorry to hear that, this sounds like something that shouldn't happen, even when downgrading.

I assume you have SSH access to your node. In our docs you can find a reference on how to access the database content: https://docs.btcpayserver.org/Docker/#how-can-i-connect-to-the-database

This way you could at least check if the old data is still present.

Other than that I'd recommend contacting myNode and also report that problem with them — maybe they can offer more insight what might have happened when downgrading.
